### PR TITLE
Share button to job details

### LINKED ISSUE #2570: Add share button to job posting detail
If someone knows someone or wants to save the posting elsewhere, we should make it easy to access the job posting url.

### PR DESCRIPTION
**Pull request by drewvolz**
Closes #2570 

Presents a platform-native share button. The link to the job posting is the shared content.

Platform | Detail | Share
--|--|--
Android | android-share-item | android-share-sheet
iOS | ios-share-item | ios-share-sheet

### CODE REVIEW

File: source/views/sis/student-work/detail.android.js [hunk 1]
```diff
@@ -137,6 +138,16 @@ function LastUpdated({when}: {when: string}) {
 	) : null
 }
 
+function shareJob(job: JobType) {
+	const jobBaseUrl =
+		'https://www.stolaf.edu/apps/stuwork/index.cfm?fuseaction=Details&jobID='
+	Share.share({
+		message: `${jobBaseUrl}${job.id}`,
+	})
+		.then(result => console.log(result))
```
Comment (hawkrives) [inline source/views/sis/student-work/detail.android.js]: Remove the logging please

**Review by hawkrives — DISMISSED** [review]

File: source/views/sis/student-work/detail.android.js (hunk 1 above)
Comment (drewvolz) [reply source/views/sis/student-work/detail.android.js]: If that's the case, we should remove it from calendar, too.

https://github.com/StoDevX/AAO-React-Native/blob/97b5a4e38e1aa2f506ef0c1f134838a3820d903c/source/views/calendar/calendar-util.js#L17-L19

Comment (hawkrives) [reply source/views/sis/student-work/detail.android.js]: Yes, we should, but in another PR.

File: source/views/sis/student-work/detail.android.js [hunk 2]
```diff
@@ -137,6 +138,14 @@ function LastUpdated({when}: {when: string}) {
 	) : null
 }
 
+function shareJob(job: JobType) {
```
Comment (hawkrives) [inline source/views/sis/student-work/detail.android.js]: One more - can you pull shareJob into a separate file, please? I dislike having the exact same thing in two places.

**Review by hawkrives — DISMISSED** [review]
Yay!

File: source/views/sis/student-work/detail.android.js (hunk 2 above)
Comment (drewvolz) [reply source/views/sis/student-work/detail.android.js]: Done.

File: source/views/sis/student-work/lib.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+// @flow
+import {Share} from 'react-native'
+import type {JobType} from './types'
+
+export function shareJob(job: JobType) {
+	const jobBaseUrl =
+		'https://www.stolaf.edu/apps/stuwork/index.cfm?fuseaction=Details&jobID='
```
Comment (hawkrives) [inline source/views/sis/student-work/lib.js]: Please use querystring or query-string to build this URL.

**Review by hawkrives — DISMISSED** [review]

**Review by hawkrives — APPROVED** [review]

**Pull request merged**